### ISSUE TITLE
Firefox 57+

### ISSUE THREAD

**Issue by WPennock** [hpaolini/procon#29]
Will Procon be compatible with the new Firefox? I haven't seen any updates on this, and I believe Firefox updates on 14 November.

**Comment by hpaolini** [comment]
no, procon will not be compatible for now. might change in the future.

**Comment by WPennock** [comment]
Thank you for letting us know and for all the work you have done on this project; it is an excellent extension. Are there any good alternatives you would recommend for Firefox 57+?

**Comment by WPennock** [comment]
Thank you for letting us know and for all the work you have done on this project; it is an excellent extension. Are there any good alternatives you would recommend for Firefox 57+?

**Comment by hpaolini** [comment]
At the moment there is no API to protect the add-on from being disabled/removed. Password-protection should be a _must_ in a content filter; therefore, releasing a half-baked version of ProCon without it would not be as convenient.

If there is enough interest in the community, I might take some time to fully port it to a Web Extension.

**Comment by mirasti** [comment]
dear hpaolini
i used to be porn addict and by the mercy of God,I found your extension and it helped me to get rid of that to some extent.I just want to recall you,the value of your work for the God and the newly believers  all around the world .please continue supporting "procon latte" by upgrading it to the new firefox version 57.
I hope one day i have enough money to donate this divine project for the sake of Christ.

**Comment by yassine944** [comment]
Hello, I have no skrills in programming, but if you still plan to create the new version maybe you (or someone else) can write a password protection application that reinstalls the addon if it is uninstalled by clicking on the firefox button and only uninstalled by the application itself after typing the password. I don't know if it's a good idea but it could be a temporary solution

**Comment by dayentech** [comment]
If you would please do what you can to make this work with quantum.  I understand you are having problems with the passwords but something functioning would be great!  I use this to block my son from sites and preventing him from using keywords he uses to look up things I don't want him to see.

**Comment by gipi85** [comment]
As hllwdfr wrote, I switched to WaterFox too, in which "Legacy" extensions work correctly. Moreover, the developer of Waterfox plans to keep them as an important part of his project, as you can read here:
https://github.com/MrAlex94/Waterfox/issues/267

Then it will be great if you could continue to test Procon Latte for the newer versions of Waterfox, which will be based on Firefox 57+, but will allow the extensions like Procon Latte to work.
Thanks

**Comment by Etcheguaray** [comment]
How to get Pro Con to work in Waterfox?? I've spent half my day trying everything I can think of! Help--Paypal donation to anyone who can solve this for me...

**Comment by gipi85** [comment]
hi @Etcheguaray , follow these easy steps (they are probably the same things you'd done with pre-Quantum versions of Firefox). :-)
1. Download the latest version of Waterfox from https://www.waterfoxproject.org/en-US/ and install it with the standard installation. (If you do the whole procedure and the add-on doesn't work, please fully uninstall Waterfox and re-install it).
2. Start Waterfox, and open the "options" menu by clicking on the button with the three lines. Select "add ons". 
3. In the page that will show up, check for the white search box, which is in the high right part of the page. Write into the box the exact name of the add-on, which is "Procon latte" and press enter to search.
4. A new page will show up with the result of your search, and you will find Mr. Paolini great software!!! ;-) Click on it and on the button to install it. After the installation is complete, close and restart Waterfox.
5. Back on Waterfox, open the options menu and check for the Procon latte among the add ons: in the page that will show up you will be able to set all the preferences you like!

6. Give a big thanks to Mr. Paolini @hpaolini  (developer of Procon latte), to Mr. Kontos @MrAlex94 (developer of Waterfox) and maybe support their work with Paypal donations. As for me I'll be fine with a thanks for writing this guide, but if you really want you can donate to me too :-P 
Together we can help making the internet a better place for everyone!

**Comment by onguarde** [comment]
I'm still using Procon filter, just stay on older version of firefox. 56.0. 
It still works on both android and the desktop!

Firefox 56.0.2 download,
https://ftp.mozilla.org/pub/firefox/releases/56.0.2/     (Windows)
https://www.apkmirror.com/apk/mozilla/firefox/firefox-56-0-release/      (Android)

Procon latte filter has been removed from Firefox addon webpage. You can download it from here,
[link removed -- installation xpi files are in this project's repo ]

Note: above will work on android too. Just open the install.html
This is workaround, details in here,
https://stackoverflow.com/questions/32615103/how-to-install-my-xpi-on-firefox-android

Let me know if you need any help setting this up.

**Comment by stephanelorrain** [comment]
I use Waterfox.

**Comment by grahamperrin** [comment]
> get Pro Con to work in Waterfox?

Simplest: 

- https://ca-archive.biz.tm/storage/1/1803/procon_latte_content_filter-4.2.0-an+fx.xpi

The link is taken from the Classic Add-ons Archive extension, which is highly recommended: 

- https://github.com/JustOff/ca-archive
